includes/wikia/Wikia.php

```diff
@@ -1106,6 +1106,34 @@ static public function onSkinTemplateOutputPageBeforeExec(SkinTemplate $skinTemp
 		$tpl->set( 'thisquery', $skinTemplate->thisquery );
 
 		$robotPolicy = Wikia::getEnvironmentRobotPolicy( $skinTemplate->getRequest() );
+		$request = $skinTemplate->getRequest();
+		$setNofollow = false;
+		if( $title->inNamespaces( NS_SPECIAL, NS_USER, NS_USER_TALK, NS_HELP) ) {
```

Comment:
What about `$wgRobotsTxtCustomRules` than can be customized per-wiki via WikiFactory interface?

Reply:
Also - if some of the wikis used `wgRobotsTxtCustomRules` to block namespaces handled by mobile-wiki, shouldn't we propagate it to mobile somehow (maybe through `specialRobotPolicy`)?

Reply:
With regard to wgRobotsTxtCustomRules I'm forcing the additional noindex only on those types of pages covered in the ticket(and they aren't generated mobile-wiki side). As far as the ticket goes it's all about keeping the same behavior as before (the noindex rules still applied even if someone set it to allow). I'll take a look at the specialRobotPolicy

Reply:
It seems that mobile-wiki only gets the default robots context for the whole domain. I can't find any place that fetches the robot policy for any given namespace. We can add this feature in the future but in terms of this ticket I dont think it's needed

Reply:
I don't think you're keeping the same behavior as before:

- for example on community wiki the `wgRobotsTxtCustomRules` is set to an ampty array which means User namespace is indexed (there is no Noindex: /wiki/User: rule in https://community.fandom.com/robots.txt). Your PR block User namespace on every wiki
- if I define wgRobotsTxtCustomRules on my wiki to block NS_FILE, am I going to get noindex meta tag on File pages?